psamm/database.py

```diff
@@ -164,28 +164,67 @@ class DictDatabase(MetabolicDatabase):
 
     def __init__(self):
         super(DictDatabase, self).__init__()
+        self._model_name = None
+        self._biomass_reaction = None
+        self._default_flux_limit = None
+        self._extracellular_compartment = None
+        # _flux_limits is a map from reaction_id to (upper, lower)
+        self._flux_limits = {}
+        # _medium is a map from reaction_id to (Compound, upper, lower)
+        self._medium = {}
+        # reaction IDs in the model
+        self._model = set()
+        # compound map from compound ID to CompoundEntry
+        self._compound_entries = defaultdict(dict)
+        # reaction map from reaction ID to ReactionEntry
+        self._reaction_entries = defaultdict(dict)
+        # original attributes in DictDatabase, consider to be moved
         self._reactions = defaultdict(dict)
         self._compound_reactions = defaultdict(set)
         self._reversible = set()
 
     @property
+    def model_name(self):
+        return self._model_name
+
+    @property
+    def extracelluar_compartment(self):
+        return self._extracellular_compartment
+
+    @property
+    def defalut_flux_limit(self):
+        return self._default_flux_limit
+
+    @property
     def reactions(self):
         return iter(self._reactions)
 
     @property
+    def compound_entries(self):
+        return iter(self._compound_entries)
+
+    @property
     def compounds(self):
         return iter(self._compound_reactions)
 
     @property
+    def model(self):
+        return iter(self._model)
+
+    # consider to be moved
+    @property
     def compartments(self):
         compartment_set = set()
         for compound in self.compounds:
             if compound.compartment not in compartment_set:
                 compartment_set.add(compound.compartment)
                 yield compound.compartment
 
+    def has_compound(self, compound_id):
+        return compound_id in self._compound_entries
+
     def has_reaction(self, reaction_id):
-        return reaction_id in self._reactions
+        return reaction_id in self._reaction_entries
```

Comment:
This change would cause some tests fail, as it doesn't have exchange reaction in _reaction_entries